### PR TITLE
Added links to the mosaic docs in figure and pyplot module docstrings

### PR DESCRIPTION
**Pull request by sveneschlbeck**
#21532 @timhoffm

## PR Summary

Added links to the provisional API documentation for the subplot_mosaic function in both the ``figure`` and ``pyplot`` module docstrings

## PR Checklist

- [ N/A] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ N/A] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ N/A] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ N/A] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by tacaswell** [comment]
Something is not right with the references: 

```
/home/circleci/project/lib/matplotlib/pyplot.py:docstring of matplotlib.pyplot.subplot_mosaic:11: WARNING: unknown document: /tutorials/provisional/mosaic.py
/home/circleci/project/lib/matplotlib/figure.py:docstring of matplotlib.figure.FigureBase.subplot_mosaic:11: WARNING: unknown document: /tutorials/provisional/mosaic.py
/home/circleci/project/lib/matplotlib/figure.py:docstring of matplotlib.figure.FigureBase.subplot_mosaic:11: WARNING: unknown document: /tutorials/provisional/mosaic.py
/home/circleci/project/lib/matplotlib/figure.py:docstring of matplotlib.figure.FigureBase.subplot_mosaic:11: WARNING: unknown document: /tutorials/provisional/mosaic.py
```

Our doc build fails on warnings which helps us avoid things like dead or broken links!

Can you also drop the trailing whitespace flagged by the CI?

**Comment by QuLogic** [comment]
The references should not have file extensions.

File: lib/matplotlib/figure.py [hunk 1]
```diff
@@ -1679,6 +1679,9 @@ def subplot_mosaic(self, mosaic, *, sharex=False, sharey=False,
            This API is provisional and may be revised in the future based on
            early user feedback.
 
+        See :doc:`/tutorials/provisional/mosaic.py` 
+        for an examplary usage and provisional API documentation
```
Comment (jklymak) [inline lib/matplotlib/figure.py:1683]: I don't think you mean "exemplary" here?

**Comment by sveneschlbeck** [comment]
@jklymak I did mean it but I changed it now in favor of a better and more understandable version of what I meant, thanks for the input

**Comment by sveneschlbeck** [comment]
@QuLogic @tacaswell I got rid of the whitespace at the end of the first line in both files and also dropped the file extensions

File: lib/matplotlib/figure.py [hunk 2]
```diff
@@ -1679,6 +1679,9 @@ def subplot_mosaic(self, mosaic, *, sharex=False, sharey=False,
            This API is provisional and may be revised in the future based on
            early user feedback.
 
+        See :doc:`/tutorials/provisional/mosaic`
+        for an example of its usage and provisional API documentation
+
```
Comment (jklymak) [inline lib/matplotlib/figure.py:1684]: I would just say "... for an example."  Or if you really want.  ... for an example and full API explanation"

Comment (sveneschlbeck) [reply lib/matplotlib/figure.py]: Done :)

**Review by jklymak — APPROVED** [review]

**Comment by jklymak** [comment]
Thanks @sveneschlbeck  (BTW, I am going to squash merge, which means your commits will be merged into a new commit, if you are tracking them somehow).

**Comment by sveneschlbeck** [comment]
Thanks for the info @jklymak

**Comment by tacaswell** [comment]
@sveneschlbeck Congratulations on what I think is your first Matplotlib PR 🎉   Hopefully we will hear from you again!

**Comment by sveneschlbeck** [comment]
@tacaswell Thank you very much! I hope so, too :)

**Pull request merged**